src/daft-table/src/ops/agg.rs

```diff
@@ -19,6 +19,21 @@ impl Table {
     }
 
     pub fn agg_groupby(&self, to_agg: &[Expr], group_by: &[Expr]) -> DaftResult<Table> {
+        let agg_exprs = to_agg
+            .iter()
+            .map(|e| match e {
+                Expr::Agg(e) => Ok(e),
+                _ => Err(DaftError::ValueError(format!(
+                    "Trying to run non-Agg expression in Grouped Agg! {e}"
+                ))),
+            })
+            .collect::<DaftResult<Vec<_>>>()?;
+
+        #[cfg(feature = "python")]
+        if let Some(AggExpr::MapGroups { func, inputs }) = agg_exprs.first() {
```

Comment:
Seems like also it's a stronger assumption that we have to make here?

```
if agg_exprs.length() == 1 && let Some(AggExpr::MapGroups { func, inputs }) = agg_exprs.first() {...}
```

Reply:
I guess we have the implicit assumption here that MapGroup will only happen when the size of the exprs is 1, since that's controlled via the user API?